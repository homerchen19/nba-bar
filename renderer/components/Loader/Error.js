import React, { PureComponent } from 'react';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '@styles/theme';

const Wrapper = styled(Flex)`
  flex: 1;
  width: 100vh;
  height: 100%;
`;

const Text = styled.p`
  margin: 5px 0;
  color: ${colors.red};
  font-size: 15px;
  font-weight: 500;
`;

const Report = styled.a`
  color: ${colors.blue};
  font-size: 15px;
  font-weight: 400;
  cursor: pointer;

  &:hover {
    margin-bottom: -1px;
    border-bottom: 1px solid ${colors.darkBlue};
    color: ${colors.darkBlue};
  }
`;

class Error extends PureComponent {
  render() {
    const {
      remote: { app },
      shell,
    } = window.require('electron');

    return (
      <Wrapper justify="center" direction="column">
        <Text>
          Oops! Something goes wrong{' '}
          <span role="img" aria-label="Sad">
            üôÅ
          </span>
        </Text>
        <Text>
          <Report
            onClick={() => {
              app.hide();
              shell.openExternal('https://github.com/xxhomey19/nba-bar/issues');
            }}
          >
            Report issues
          </Report>
        </Text>
      </Wrapper>
    );
  }
}

export default Error;
