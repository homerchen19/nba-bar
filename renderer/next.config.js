const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
