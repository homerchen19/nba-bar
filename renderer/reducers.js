import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
