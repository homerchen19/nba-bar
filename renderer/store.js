import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
