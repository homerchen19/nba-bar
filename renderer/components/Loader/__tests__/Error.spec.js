import React from 'react';
import renderer from 'react-test-renderer';

import Error from '../Error';

it('be defined', () => {
  expect(Error).toBeDefined();
});

it('match snapshot', () => {
  window.require = jest.fn(() => ({
    remote: {
      app: { hide: jest.fn() },
    },
    shell: {
      openExternal: jest.fn(),
    },
  }));

  const tree = renderer.create(<Error />).toJSON();

  expect(tree).toMatchSnapshot();
});
