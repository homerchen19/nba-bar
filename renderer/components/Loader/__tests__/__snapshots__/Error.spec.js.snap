// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox Error__Wrapper-sc-1ln6l78-0 fnepRI am-flexbox-dir-column am-flexbox-justify-center am-flexbox-align-center"
>
  <p
    className="Error__Text-sc-1ln6l78-1 cwwiZS"
  >
    Oops! Something goes wrong
     
    <span
      aria-label="Sad"
      role="img"
    >
      🙁
    </span>
  </p>
  <p
    className="Error__Text-sc-1ln6l78-1 cwwiZS"
  >
    <a
      className="Error__Report-sc-1ln6l78-2 fPDaPQ"
      onClick={[Function]}
    >
      Report issues
    </a>
  </p>
</div>
`;
